storage/innobase/trx/trx0rseg.cc

```diff
@@ -127,17 +127,13 @@ void trx_rseg_update_wsrep_checkpoint(const XID* xid)
 @return	whether the WSREP XID was present */
 bool trx_rseg_read_wsrep_checkpoint(const trx_rsegf_t* rseg_header, XID& xid)
 {
-	xid.formatID = (int)mach_read_from_4(
+	int formatID = (int)mach_read_from_4(
```

Comment:
I would prefer to avoid type casts, especially C-style ones.

Reply:
Agreed. However, I chose to follow the style used in the surrounding code. Not having a cast might emit a warning from compiler if -Wsign-conversion is enabled.

I can change this to static cast and fix the surrounding wsrep related code too, would that be ok?

Reply:
Yes, please feel free to fix. I also just moved the old code around without removing the type conversion. I think that the `formatID` could be unsigned; in fact, I understood that Galera would only write it as 1. Maybe other values than 1 should be ignored?